src/scripts/wind_speed_retrieval.py

```diff
@@ -0,0 +1,182 @@
+        apply_flattening = True
+        GMF_VH_NAME = "gmf_s1_v2"
+    else : 
+        print("mission_name should be SENTINEL-1 since we only know to use S1")
+        apply_flattening = False
+        GMF_VH_NAME = "gmf_rs2_v2"
+        return 
+
+    nesz_cr = xsar_obj_1000m.dataset.nesz.isel(pol=1) #(no_flattening)
+    if apply_flattening : 
+        xsar_obj_1000m.dataset=xsar_obj_1000m.dataset.assign(nesz_VH_final=(['line','sample'],windspeed.nesz_flattening(nesz_cr, xsar_obj_1000m.dataset.incidence)))
+        xsar_obj_1000m.dataset['nesz_VH_final'].attrs["comment"] = 'nesz has been flattened using windspeed.nesz_flattening'
+    else :
+        xsar_obj_1000m.dataset=xsar_obj_1000m.dataset.assign(nesz_VH_final=(['line','sample'],nesz_cr.values))
+        xsar_obj_1000m.dataset['nesz_VH_final'].attrs["comment"] = 'nesz has not been flattened'
+   
+    ## dsig
+    try : 
+        dsig_cr = windspeed.get_dsig("gmf_s1_v2", dataset_1000m.incidence,dataset_1000m.sigma0_ocean.sel(pol='VH'),dataset_1000m.nesz_VH_final)
+    except Exception as e :
+        print(e)
+        return
+
+    ## co & dual inversion
+    windspeed_co, windspeed_dual = windspeed.invert_from_model(
+        xsar_obj_1000m.dataset.incidence,
+        xsar_obj_1000m.dataset.sigma0_ocean.isel(pol=0),
+        xsar_obj_1000m.dataset.sigma0_ocean.isel(pol=1),
+        #ancillary_wind=-np.conj(xsar_obj_1000m.dataset['ancillary_wind']),
+        ancillary_wind=-xsar_obj_1000m.dataset['ancillary_wind'],
+        dsig_cr = dsig_cr,
+        model=('cmod5n',GMF_VH_NAME))
+        
+    xsar_obj_1000m.dataset["windspeed_co"] = np.abs(windspeed_co)
+    xsar_obj_1000m.dataset["windspeed_co"].attrs["comment"] = xsar_obj_1000m.dataset["windspeed_co"].attrs["comment"].replace("wind speed and direction","wind speed")
+
+    xsar_obj_1000m.dataset["windspeed_dual"] = np.abs(windspeed_dual)
+    xsar_obj_1000m.dataset["windspeed_dual"].attrs["comment"] = xsar_obj_1000m.dataset["windspeed_dual"].attrs["comment"].replace("wind speed and direction","wind speed")
+    
+    ## cr inversion ##TODO
+    windspeed_cr = windspeed.invert_from_model(
+        xsar_obj_1000m.dataset.incidence.values,
+        xsar_obj_1000m.dataset.sigma0_ocean.isel(pol=1).values,
+        #ancillary_wind=-np.conj(xsar_obj_1000m.dataset['ancillary_wind']),
+        dsig_cr = dsig_cr.values,
+        model=GMF_VH_NAME)
+
+    windspeed_cr = np.abs(windspeed_cr)
+    xsar_obj_1000m.dataset=xsar_obj_1000m.dataset.assign(windspeed_cr=(['line','sample'],windspeed_cr))
+    xsar_obj_1000m.dataset.windspeed_cr.attrs['comment'] = "wind speed inverted from model %s (%s)" % (GMF_VH_NAME, "VH")
+    xsar_obj_1000m.dataset.windspeed_cr.attrs['model'] = GMF_VH_NAME
+    xsar_obj_1000m.dataset.windspeed_cr.attrs['units'] = 'm/s'
+    
+    
+    ## saving 
+    
+    # prepare dataset for netcdf export
+    black_list = ['model_U10', 'model_V10', 'digital_number', 'gamma0_raw', 'negz',
```

Comment:
is a "white" list more suitable than a black list here?

Reply:
no it is black list since they are the variable to be removed